tests/python/test_dynamic_append_length.py

```diff
@@ -0,0 +1,36 @@
+import taichi as ti
+from tests import test_utils
+
+
+def _test_dynamic_append_length(dt):
+    x = ti.field(int)
+    block = ti.root.dense(ti.i, 10)
+    pixel = block.dynamic(ti.j, 10)
+    pixel.place(x)
+
+    @ti.kernel
+    def test():
+        for i in range(10):
+            for j in range(i):
+                x[i].append(j)
+        for i in range(10):
+            assert (ti.length(x.parent(), i) == i)
+            for j in range(i):
+                assert (x[i, j] == j)
+
+    test()
+
+
+@test_utils.test(exclude=[ti.cc, ti.opengl, ti.vulkan],
+                 default_fp=ti.f32,
+                 debug=True)
+def test_dynamic_append_length_f32():
+    _test_dynamic_append_length(ti.f32)
+
+
+@test_utils.test(exclude=[ti.cc, ti.opengl, ti.vulkan],
+                 default_fp=ti.f64,
+                 require=ti.extension.data64,
+                 debug=True)
+def test_dynamic_append_length_f64():
+    _test_dynamic_append_length(ti.f64)
```

Comment:
```suggestion
```

I don't think `ti.f64` works here - the test passes likely by luck. See 
https://github.com/taichi-dev/taichi/blob/5b69178e2ee45a2f2fe9ecea786bd61356f0a8d5/taichi/runtime/llvm/runtime_module/node_dynamic.h#L61